rules_building_block/execution_github_high_number_of_cloned_repos_from_pat.toml

```diff
@@ -0,0 +1,59 @@
+[metadata]
+creation_date = "2023/10/11"
+integration = ["github"]
+maturity = "production"
+min_stack_comments = "New fields added to GitHub Integration"
+min_stack_version = "8.8.0"
+updated_date = "2023/12/14"
+bypass_bbr_timing = true
+
+[rule]
+author = ["Elastic"]
+building_block_type = "default"
+description = """
+Detects a high number of unique private repo clone events originating from a single personal access token within a short time period.
+"""
+from = "now-6m"
+index = ["logs-github.audit-*"]
+language = "kuery"
+license = "Elastic License v2"
+name = "High Number of Cloned GitHub Repos From PAT"
+risk_score = 21
+rule_id = "fb0afac5-bbd6-49b0-b4f8-44e5381e1587"
+severity = "low"
+tags = ["Domain: Cloud",
+        "Use Case: Threat Detection",
+        "Use Case: UEBA",        
+        "Tactic: Execution",
+        "Rule Type: BBR",
+        "Data Source: Github"
+        ]
+timestamp_override = "event.ingested"
+type = "threshold"
+
+query = '''
+event.dataset:"github.audit" and event.category:"configuration" and event.action:"git.clone" and 
+github.programmatic_access_type:("OAuth access token" or "Fine-grained personal access token") and 
+github.repository_public:false
```

Comment:
Nice! Yea this would be sort of odd. Unless it is a CI job or sorts that relies on multiple repositories and uses a single PAT for access. 

If we managed to see OAuth that would be even more odd considering consent screen requirements. I'd vote that this could potentially be a normal rule itself instead of a BBR. A compromised access token could definitely be used to do such activity.

This may fall more under https://attack.mitre.org/techniques/T1078/004/ as well for technique since it likely involves access token compromise for a legitimate account.

Reply:
Sure! I'll keep an eye on this one in telemetry though